src/main/java/mate/academy/springboot/datajpa/service/impl/ProductServiceImpl.java

```diff
@@ -0,0 +1,48 @@
+package mate.academy.springboot.datajpa.service.impl;
+
+import java.math.BigDecimal;
+import java.util.List;
+import mate.academy.springboot.datajpa.model.Category;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.repository.ProductRepository;
+import mate.academy.springboot.datajpa.service.ProductService;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ProductServiceImpl implements ProductService {
+    private final ProductRepository productRepository;
+
+    public ProductServiceImpl(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    @Override
+    public Product add(Product model) {
+        return productRepository.save(model);
+    }
+
+    @Override
+    public Product get(Long id) {
+        return productRepository.getReferenceById(id);
```

Comment:
This method(getRefernceByid()) returns a proxy before going to DB and fetching the entity. The problem is if there is no entity - an exception will be thrown (the same will happens with getById() too), so in my opinion, better to use findById() cos it returns an Optional<>
But of course, it's up to you to choose what option would be better for your solution.

Reply:
I use it because getById() is deprecated and the use of getReferenceById is specified in the documentation
![image](https://github.com/mate-academy/jv-springboot-data-jpa/assets/92263491/e8768bbf-71ea-4cb7-bfca-bfa6a058590b)